snuba/datasets/configuration/utils.py

```diff
@@ -109,60 +112,50 @@ def get_mandatory_condition_checkers(
 
 def __parse_simple(
     col: dict[str, Any], modifiers: SchemaModifiers | None
-) -> Column[SchemaModifiers]:
-    return Column(col["name"], SIMPLE_COLUMN_TYPES[col["type"]](modifiers))
+) -> ColumnType[SchemaModifiers]:
+    return SIMPLE_COLUMN_TYPES[col["type"]](modifiers)
 
 
 def __parse_number(
     col: dict[str, Any], modifiers: SchemaModifiers | None
-) -> Column[SchemaModifiers]:
-    return Column(
-        col["name"], NUMBER_COLUMN_TYPES[col["type"]](col["args"]["size"], modifiers)
-    )
+) -> ColumnType[SchemaModifiers]:
+    col_type = NUMBER_COLUMN_TYPES[col["type"]]
+    # UInt and Float ColumnType accepts a size parameter
+    return col_type(col["args"]["size"], modifiers)  # type: ignore
```

Comment:
Why the type ignore here?

Reply:
Base `ColumnType` does not accept a `size` parameter like `Float` and `UInt` do so mypy complains that `ColumnType` was given too many parameters since `NUMBER_COLUMN_TYPES` is a `dict[str, ColumnType]`. I could add some assertion that `col_type` is a `Float` or `UInt` instead of the type ignore if that's a better way to do this.

Reply:
Yeah I would rather do that.